bin/si-sdf-api-test/tests/8-check_mjolnir.ts

```diff
@@ -0,0 +1,187 @@
+  }
+
+  const createComponentResp = await sdfApiClient.call({
+    route: "create_component",
+    body: createComponentPayload,
+  });
+
+  const newComponentId = createComponentResp?.componentId;
+  assert(newComponentId, "Expected to get a component id after creation");
+  
+  // Check that component exists on diagram
+  const diagram = await sdfApiClient.call({
+    route: "get_diagram",
+    routeVars: { changeSetId },
+  });
+  assert(diagram?.components, "Expected components list on the diagram");
+  assert(
+    diagram.components.length === 1,
+    "Expected a single component on the diagram",
+  );
+  const createdComponent = diagram.components[0];
+  assert(
+    createdComponent?.id === newComponentId,
+    "Expected diagram component id to match create component API return ID",
+  );
+  assert(
+    createdComponent?.schemaVariantId === schemaVariantId,
+    "Expected diagram component schema variant id to match sv id",
+  );
+
+  // Get the checksum for the individual component MV
+  let maybeChecksum = null;
+  for (let count = 1; count < INDEX_MAX_CALLS - 1; count++) {
+    const response = await sdfApiClient.call({
+      route: "index",
+      routeVars: { changeSetId },
+    });
+
+    // If we find the checksum, then we are done
+    for (const mv of response?.frontEndObject?.data?.mvList) {
+      if ((mv?.id === newComponentId) && (mv?.kind === "Component")) {
+        maybeChecksum = mv?.checksum;
+        break;
+      }
+    }
+    if (maybeChecksum) {
+      break;
+    }
+      
+    console.log(`index not yet built or available (sleeping for ${INDEX_SLEEP_MS}ms: attempt ${count} of ${INDEX_MAX_CALLS})`);
+    await new Promise(f => setTimeout(f, INDEX_SLEEP_MS));
+  }
+
+  // Confirm that we have the checksum
+  const checksum = maybeChecksum!;
+
+  // With the checksum in hand, get the component MV
+  let maybeComponentName = null;
+  for (let count = 1; count < MJOLNIR_MAX_CALLS - 1; count++) {
+    const response = await sdfApiClient.call({
+      route: "mjolnir",
+      routeVars: { changeSetId, materializedViewId: newComponentId, referenceKind: "Component", materializedViewChecksum: checksum },
+    }, true);
+
+    // If the call succeeded, try to get the component name
+    if (response?.status === 200) {
+      try {
+        const json = await response.json();
+        maybeComponentName = json?.frontEndObject?.data?.name;
+      } catch (err) {
+        console.error("Error trying to parse response body as JSON", err);
+      }
+      
+      // If we found the component name, then we are done (otherwise, retry!)
+      if (maybeComponentName) {
+        break;
+      }
+    } else if (response?.status === 404) {
+      // Retry on 404
+      console.log(`Received 404 error (retrying): ${await response.text()}`);
+    } else {
+      // Fail on non-200 and non-404 errors
+      throw new Error(`Error ${response.status}: ${await response.text()}`);
+    }
```

Comment:
This is a complex scenario but based on the result of the index, we want varying levels of logging and error handling. This likely changes as we harden this layer.